src/extension.ts

```diff
@@ -317,7 +317,7 @@ async function launchMain(mainMethods: IMainClassOption[], uri: vscode.Uri, noDe
         return;
     }
 
-    await startDebugging(pick.mainClass, pick.projectName, uri, noDebug);
+    await startDebugging(pick.mainClass, pick.projectName!, uri, noDebug);
```

Comment:
pick.projectName!

What is the purpose of the ending **!**

Reply:
Because the second parameter of `startDebugging` could not be undefined, but `pick.projectName` could be, since it's  optional. How about change projectName in `IMainClassOption` to required property?

Reply:
https://www.typescriptlang.org/docs/handbook/release-notes/typescript-2-0.html#non-null-assertion-operator

x! will throw exception if x is null or undefined. This is not we want. I would change it to 
`await startDebugging(pick.mainClass, pick.projectName || "", uri, noDebug);`

Reply:
OK. BTW, I didn't do this because I don't know what is a proper default value.